Calc

SergeySkosyrskiy opened the pull request.

SergeySkosyrskiy committed:
Первая проба

SergeySkosyrskiy committed:
Конечный вариант

pavelilyukevichsky commented on src/main/java/pro/sky/calculator/CalController.java:
```diff
@@ -0,0 +1,66 @@
+        this.calcService = calcService;
+    }
+
+    @GetMapping(path = "/calculator")
```
если один параметр то можно не указывать path =

pavelilyukevichsky commented on src/main/java/pro/sky/calculator/CalController.java:
```diff
@@ -0,0 +1,66 @@
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping()
```
"/calculator" забыли добавить. и Тогда не надо будет дублировать в других методах "/calculator"
будет просто "/plus"

SergeySkosyrskiy replied:
В задании написано 
1. Метод по адресу /calculator должен вернуть приветствие “Добро пожаловать в калькулятор".

pavelilyukevichsky commented on src/main/java/pro/sky/calculator/CalController.java:
```diff
@@ -0,0 +1,66 @@
+            return "Ошибка!!! Введены не все данные!!!";
+        } else {
+            int result = calcService.sum(num1, num2);
+            return num1 + "+" + num2 + "=" + (num1 + num2);
```
грубая ошибка. у нас же есть результат в переменной result. заме нам сложение делать в 29 строке?
calcService должен управлять вычислениями. контроллер только контролирует входные параметры и отдает результат в виде строки.

pavelilyukevichsky commented on src/main/java/pro/sky/calculator/CalController.java:
```diff
@@ -0,0 +1,66 @@
+            return "Ошибка!!! Введены не все данные!!!";
+        } else {
+            int result = calcService.subtract(num1, num2);
+            return num1 + "-" + num2 + "=" + (num1 - num2);
```
таже проблема

pavelilyukevichsky commented on src/main/java/pro/sky/calculator/CalController.java:
```diff
@@ -0,0 +1,66 @@
+            return "Ошибка!!! Введены не все данные!!!";
+        } else {
+            int result = calcService.multiply(num1, num2);
+            return num1 + "*" + num2 + "=" + (num1 * num2);
```
таже проблема

pavelilyukevichsky commented on src/main/java/pro/sky/calculator/CalController.java:
```diff
@@ -0,0 +1,66 @@
+            return "ОШИБКА!!! на 0 делить нельзя!!!";
+        } else {
+            int result = calcService.divide(num1, num2);
+            return num1 + "/" + num2 + "=" + (num1 / num2);
```
таже проблема

pavelilyukevichsky commented on src/main/java/pro/sky/calculator/CalcService.java:
```diff
@@ -0,0 +1,30 @@
+import org.springframework.web.bind.annotation.RestController;
+
+@Service
+@RestController
```
не нужная аннотация. тут только `@Service` нужна.
название класса CalculatorServiceImpl должно быть.

SergeySkosyrskiy committed:
В задании написано

1. Метод по адресу /calculator должен вернуть приветствие “Добро пожаловать в калькулятор".

pavelilyukevichsky commented on src/main/java/pro/sky/calculator/CalController.java:
```diff
@@ -6,61 +6,71 @@
 
 
-    @GetMapping(path = "/calculator/multiply")
+//    @GetMapping("/multiply")
```
вот это надо было раскомментировать. все верно сделали.
